Servers/UI/OJS.Servers.Ui/ClientApp/src/hooks/submissions/use-contest-categories-breadcrumb.tsx

```diff
@@ -0,0 +1,105 @@
+﻿import React, { createContext, useCallback, useContext, useState } from 'react';
+import { isNil } from 'lodash';
+import { IContestCategoryTreeType } from '../../common/types';
+import { ITreeItemType } from '../../components/guidelines/trees/Tree';
+import { IHaveChildrenProps } from '../../components/common/Props';
+
+interface ICategoriesBreadcrumbContext {
+    state: {
+        breadcrumbItems: ICategoriesBreadcrumbItem[];
+    };
+    actions: {
+        updateBreadcrumb: (category: ITreeItemType | undefined, flattenTree: ITreeItemType[] | []) => void;
+    };
+}
+
+interface ICategoriesBreacrumbProviderProps extends IHaveChildrenProps {
+}
+
+interface ICategoriesBreadcrumbItem {
+    id: string,
+    isLast: boolean,
+    value: string,
+    orderBy: number,
+}
+
+const defaultState = { state: { breadcrumbItems: [] as ICategoriesBreadcrumbItem[] } };
+
+const CategoriesBreadcrumbContext = createContext<ICategoriesBreadcrumbContext>(defaultState as ICategoriesBreadcrumbContext);
+
+const orderByAsc = (x : ICategoriesBreadcrumbItem, y: ICategoriesBreadcrumbItem) => y.orderBy - x.orderBy;
+
+const CategoriesBreadcrumbProvider = ({ children }: ICategoriesBreacrumbProviderProps) => {
+    const [ breadcrumbItems, setBreadcrumbItems ] = useState(defaultState.state.breadcrumbItems);
+
+    const updateBreadcrumb = useCallback(
+        (category: IContestCategoryTreeType | undefined, categoriesTree: ITreeItemType[] | []) => {
+            if (isNil(category) || isNil(categoriesTree)) {
```

Comment:
`categories` can be fetched from `useContestCategories`. No need to pass them as args.

Reply:
Categories are different collection. The variable 'categories' in useContestCategories contains the first dimeson of the tree. 'categoriesTree' passed by argument contains the flattened tree of the categories.